tests/functional/test-no-lessee-id.sh

```diff
@@ -0,0 +1,141 @@
+#####################################
+
+# Tests that an owner cannot create an offer on a node that would cause a time conflict
+
+echo "OFFER CREATE TIME CONFLICT TEST"
+
+openstack --os-cloud test1 esi offer create \
+  $node_uuid \
+  --start-time $start \
+  --end-time $end \
+  --resource-type $resource_type \
+  -f shell > $tmpfile 2> $errfile
+ec=$?
+expected_error="Time conflict for dummy_node $node_uuid."
+
+if ! cat $errfile | grep -q "$expected_error"; then
+  if [[ $ec -eq 0 ]]; then
+    echo "ERROR: succeeded creating offer despite time conflict" >&2
+    . $tmpfile
+    echo "Deleting time conflict test offer..."
+    ./cleanup.sh $uuid
+  else
+    echo "ERROR: unexpected error during time conflict test" >&2
+    cat $errfile >&2
+  fi
+  exit 1
+fi
+
+echo "OFFER CREATE TIME CONFLICT TEST SUCCEEDED"
+
+#################################
+## OFFER DELETE NOT FOUND TEST ##
+#################################
+
+# Tests that an owner cannot delete a nonexistent offer
+
+echo "OFFER DELETE NOT FOUND TEST"
+
+openstack --os-cloud test1 esi offer delete "notanofferuuid" > $tmpfile 2> $errfile
+ec=$?
+expected_error='Offer with name or uuid notanofferuuid not found.'
+if ! grep -q "$expected_error" $errfile; then
+  if [[ $ec -eq 0 ]]; then
+    echo "ERROR: succeeded deleting offer that shouldn't exist. Either someone named their offer 'notanofferuuid' or the delete command gave a success when it shouldn't have" >&2
+  else
+    echo "ERROR: unexpected error during offer not found test" >&2
+    cat $errfile >&2
+  fi
+  exit 1
+fi
+
+echo "OFFER DELETE NOT FOUND TEST SUCCEEDED"
+
+############################################
+## OFFER NOT AVAILABLE IN TIME RANGE TEST ##
+############################################
+
+# Tests that a lessee cannot claim an offer for a time range it is not available for
+
+echo "OFFER NOT AVAILABLE IN TIME RANGE TEST"
+
+openstack --os-cloud test1-subproject esi offer claim $test_offer_uuid --start-time $(date -d "+6 days" +%Y-%m-%d) --end-time $(date -d "+7 days" +%Y-%m-%d) -f shell > $tmpfile 2> $errfile
+ec=$?
+expected_error="Offer $test_offer_uuid has no availabilities at given time range"
+
+if ! grep -q "$expected_error" $errfile; then
+  if [[ $ec -eq 0 ]]; then
+    echo "ERROR: offer was successfully claimed during invalid time range" >&2
+  else
+    echo "ERROR: unexpected error during not available in time range test" >&2
+    cat $errfile >&2
+  fi
+  exit 1
+fi
+
+echo "OFFER NOT AVAILABLE IN TIME RANGE TEST SUCCEEDED"
+
+######################
+## OFFER CLAIM TEST ##
+######################
+
+# Tests that a lessee can claim an offer available to them
+# Setting up a claim is needed for the claim time conflict test
+
+echo "OFFER CLAIM TEST"
+
+openstack --os-cloud test1-subproject esi offer claim $test_offer_uuid -f shell > $tmpfile \
+  || { ec=$?; echo "ERROR: failed to claim offer" >&2; exit $ec; }
+cat $tmpfile
+. $tmpfile
+
+echo "OFFER CLAIM TEST SUCCEEDED"
+
```

Comment:
I think you could also create another offer_claim_test that another lessee can claim this offer for a different time range.

Reply:
You mean only claiming it for part of the time range instead of the whole offer range?

Reply:
I suggest focusing on fixing the tests already in this PR, and moving any requests for additional tests to a new issue (and a future PR).

Reply:
> You mean only claiming it for part of the time range instead of the whole offer range?

yeah, that's what I mean. But you could ignore this new test for now as lars suggested.

Reply:
@DanNiESh could you check the test spreadsheet to see if the test you suggested is in there, and add it in if it's not? That's probably the best way to ensure a followup.

Reply:
Okay, @betaredex, I added a comment in the spreadsheet about it.